.github/workflows/artifact-tests.yml

```diff
@@ -0,0 +1,144 @@
+name: artifact-unit-tests
+on:
+  push:
+    paths:
+        - 'packages/artifact/**'
+  pull_request:
+    paths:
+        - 'packages/artifact/**'
+
+jobs:
+
+  build:
+    name: Build
+
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest]
+      fail-fast: false
+
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+
+    # Need root node_modules because certain npm packages like jest are configured for the entire repository and it won't be possible
+    # without these to just compile the artifacts package
+    - name: Install root npm packages
+      run: npm install
+
+    - name: Compile artifact package
+      run: |
+        npm install
+        tsc
```

Comment:
We should use `npm run tsc` for this so that we know we're using the correct Typescript version from `node_modules` rather than the `tsc` on the path.